op-chain-ops/genesis/config.go

```diff
@@ -338,172 +459,232 @@ func (d *DeployConfig) Check() error {
+		if d.DACommitmentType != plasma.GenericCommitmentString {
+			if d.DAChallengeWindow == 0 {
+				return fmt.Errorf("%w: DAChallengeWindow cannot be 0 when using alt-da mode with Keccak Commitments", ErrInvalidDeployConfig)
+			}
+			if d.DAResolveWindow == 0 {
+				return fmt.Errorf("%w: DAResolveWindow cannot be 0 when using alt-da mode with Keccak Commitments", ErrInvalidDeployConfig)
+			}
 		}
 	}
-	// L2 block time must always be smaller than L1 block time
-	if d.L1BlockTime < d.L2BlockTime {
-		return fmt.Errorf("L2 block time (%d) is larger than L1 block time (%d)", d.L2BlockTime, d.L1BlockTime)
-	}
+	return nil
+}
+
+// L2InitializationConfig represents all L2 configuration
+// data that can be configured before the deployment of any L1 contracts.
+type L2InitializationConfig struct {
+	DevDeployConfig
+	L2GenesisBlockDeployConfig
+	OwnershipDeployConfig
+	L2VaultsDeployConfig
+	GovernanceDeployConfig
+	GasPriceOracleDeployConfig
+	GasTokenDeployConfig
+	OperatorDeployConfig
+	EIP1559DeployConfig
+	UpgradeScheduleDeployConfig
+	L2CoreDeployConfig
+	PlasmaDeployConfig
+}
+
+func (d *L2InitializationConfig) Check(log log.Logger) error {
+	return checkConfigBundle(d, log)
+}
+
+// DevL1DeployConfig is used to configure a L1 chain for development/testing purposes.
+// A production L2 deployment does not utilize this configuration,
+// except of a L1BlockTime sanity-check (set this to 12 for L1 Ethereum).
+type DevL1DeployConfig struct {
+	L1BlockTime                 uint64         `json:"l1BlockTime"`
+	L1GenesisBlockTimestamp     hexutil.Uint64 `json:"l1GenesisBlockTimestamp"`
+	L1GenesisBlockNonce         hexutil.Uint64 `json:"l1GenesisBlockNonce"`
+	L1GenesisBlockGasLimit      hexutil.Uint64 `json:"l1GenesisBlockGasLimit"`
+	L1GenesisBlockDifficulty    *hexutil.Big   `json:"l1GenesisBlockDifficulty"`
+	L1GenesisBlockMixHash       common.Hash    `json:"l1GenesisBlockMixHash"`
+	L1GenesisBlockCoinbase      common.Address `json:"l1GenesisBlockCoinbase"`
+	L1GenesisBlockNumber        hexutil.Uint64 `json:"l1GenesisBlockNumber"`
+	L1GenesisBlockGasUsed       hexutil.Uint64 `json:"l1GenesisBlockGasUsed"`
+	L1GenesisBlockParentHash    common.Hash    `json:"l1GenesisBlockParentHash"`
+	L1GenesisBlockBaseFeePerGas *hexutil.Big   `json:"l1GenesisBlockBaseFeePerGas"`
+}
+
+type SuperchainL1DeployConfig struct {
+	// RequiredProtocolVersion indicates the protocol version that
+	// nodes are required to adopt, to stay in sync with the network.
+	RequiredProtocolVersion params.ProtocolVersion `json:"requiredProtocolVersion"`
+	// RequiredProtocolVersion indicates the protocol version that
+	// nodes are recommended to adopt, to stay in sync with the network.
+	RecommendedProtocolVersion params.ProtocolVersion `json:"recommendedProtocolVersion"`
+
+	// SuperchainConfigGuardian represents the GUARDIAN account in the SuperchainConfig. Has the ability to pause withdrawals.
+	SuperchainConfigGuardian common.Address `json:"superchainConfigGuardian"`
```

Comment:
What makes this go in `SuperchainL1DeployConfig` rather than `OwnershipConfig`?

Reply:
Added godocs to clarify. The OwnershipConfig is specific to the L2 chain deployment, not superchain-wide.